File: Programmers/KaKao/2019_겨울인턴십_튜플.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public int[] solution(String s) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        String origin = s.substring(1, s.length() - 1);
        String[] splited = origin.split("\\},\\{");
        
        splited[0] = splited[0].substring(1, splited[0].length());
        splited[splited.length - 1] = splited[splited.length - 1].
            substring(0, splited[splited.length - 1].length() - 1);
        
        Arrays.sort(splited, (str1, str2) -> str1.length() - str2.length());
        
        for(String str : splited) {
            String[] tuple = str.split(",");
            for(String numStr : tuple) {
                int num = Integer.parseInt(numStr);
                if(!answer.contains(num)) {
                    answer.add(num);
                }
            }
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: Programmers/KaKao/2019_겨울인턴십_튜플.kt
class Solution {
    fun solution(s: String): IntArray {
        var answer:ArrayList<Int> = ArrayList()

        var array:List<String> = s.substring(2, s.length - 2).split("},{")
        var list:ArrayList<List<String>> = ArrayList()

        for(tuple in array) {
            list.add(tuple.split(","))
        }

        var sort = list.sortedWith(Comparator<List<String>>{ a, b ->
            when {
                a.size > b.size -> 1
                a.size < b.size -> -1
                else -> 0
            }
        })

        for(tuple in sort) {
            for(num in tuple) {
                if(!answer.contains(Integer.parseInt(num))) {
                    answer.add(Integer.parseInt(num))
                }
            }
        }

        return answer.toIntArray()
    }
    
    fun solution2(s: String): IntArray {
        return s.substring(2 until s.length - 2)
            .split("},{")
            .asSequence()
            .map { it.split(",").map { num -> num.toInt() } }
            .toList()
            .sortedBy { it.size }
            .fold(setOf<Int>()) { acc, list -> acc.union(list) }
            .toIntArray()
    }
}
